server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -105,13 +119,45 @@ public RemoteFsTranslog(
         BooleanSupplier startedPrimarySupplier,
         RemoteTranslogTransferTracker remoteTranslogTransferTracker,
         RemoteStoreSettings remoteStoreSettings
+    ) throws IOException {
+        this(
+            config,
+            translogUUID,
+            deletionPolicy,
+            globalCheckpointSupplier,
+            primaryTermSupplier,
+            persistedSequenceNumberConsumer,
+            blobStoreRepository,
+            threadPool,
+            startedPrimarySupplier,
+            remoteTranslogTransferTracker,
+            remoteStoreSettings,
+            0
+        );
+    }
+
+    public RemoteFsTranslog(
+        TranslogConfig config,
+        String translogUUID,
+        TranslogDeletionPolicy deletionPolicy,
+        LongSupplier globalCheckpointSupplier,
+        LongSupplier primaryTermSupplier,
+        LongConsumer persistedSequenceNumberConsumer,
+        BlobStoreRepository blobStoreRepository,
+        ThreadPool threadPool,
+        BooleanSupplier startedPrimarySupplier,
+        RemoteTranslogTransferTracker remoteTranslogTransferTracker,
+        RemoteStoreSettings remoteStoreSettings,
+        long timestamp
     ) throws IOException {
         super(config, translogUUID, deletionPolicy, globalCheckpointSupplier, primaryTermSupplier, persistedSequenceNumberConsumer);
         logger = Loggers.getLogger(getClass(), shardId);
         this.startedPrimarySupplier = startedPrimarySupplier;
         this.remoteTranslogTransferTracker = remoteTranslogTransferTracker;
         fileTransferTracker = new FileTransferTracker(shardId, remoteTranslogTransferTracker);
```

Comment:
`fileTransferTracker` is getting convoluted now . Files are getting added when uploaded to remote store, deleted when removed from local store . If we keep the pinned tlogs files, would it really grow too large ? We are already keeping `metadataFilePinnedTimestampMap`  . Probably we could remove the pinned timestamps generation files from `fileTransferTracker` to make it light weight .

Reply:
IMO, the use of `fileTransferTracker` is to understand if local tlog and ckp files are uploaded to the remote store. We were using it to check if a particular file is present in remote store (even if it is not present in local) but with new changes, that is not required anymore. We will always be able to track files in remote store based on metadata files present.

Reply:
I agree .  Can we call this out please  as this can confuse later ? Also remove the part where it is removed on remote file deletion .

Reply:
>Can we call this out please as this can confuse later ?

Added.